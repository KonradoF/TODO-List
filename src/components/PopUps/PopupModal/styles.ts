import EStyleSheet from 'react-native-extended-stylesheet';
import normalize from 'react-native-normalize';

export const styles = EStyleSheet.create({
  firstContainer: {
    backgroundColor: 'rgba(0,0,0,0.5)',
    flex: 1,
  },
  safeArea: {flex: 0.5, justifyContent: 'flex-end'},
  secondContainer: {
    backgroundColor: '$white',
    paddingHorizontal: '5%',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    paddingBottom: '$popupModalPaddingBottom',
  },
  itemsContainer: {
    alignItems: 'flex-start',
    width: '100%',
  },
  btnContainer: {
    width: '100%',
    height: normalize(60),
    justifyContent: 'center',
  },
  btn: {
    paddingTop: normalize(10),
    paddingBottom: normalize(10),
    backgroundColor: '$primary',
  },
  closeBtn: {
    paddingTop: normalize(10),
    paddingBottom: normalize(10),
    backgroundColor: '$gray',
  },
});
