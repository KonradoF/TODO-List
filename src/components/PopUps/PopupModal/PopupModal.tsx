import React, {ReactElement} from 'react';
import {GestureResponderEvent, Modal, SafeAreaView, View} from 'react-native';
import {Button} from 'react-native-elements';
import {styles} from './styles';

interface Props {
  visible: boolean;
  onClose: () => void;
  button?: (event: GestureResponderEvent) => void;
  buttonTitle?: string;
  buttonDisabled?: boolean;
  children?: ReactElement;
}

export const PopupModal = ({
  visible,
  onClose,
  button,
  buttonTitle = 'Ukryj',
  buttonDisabled,
  children,
}: Props): ReactElement => {
  return (
    <Modal
      animationType={'fade'}
      supportedOrientations={['portrait']}
      visible={visible}
      transparent
      onRequestClose={onClose}>
      <View style={styles.firstContainer} onTouchStart={onClose} />
      <SafeAreaView style={styles.safeArea}>
        <View style={styles.secondContainer}>
          <View style={styles.itemsContainer}>{children && children}</View>
          {button && (
            <Button
              onPress={button}
              disabled={buttonDisabled}
              title={buttonTitle}
              containerStyle={styles.btnContainer}
              buttonStyle={styles.btn}
            />
          )}
          <Button
            onPress={onClose}
            title={'Close'}
            containerStyle={styles.btnContainer}
            buttonStyle={styles.closeBtn}
          />
        </View>
      </SafeAreaView>
    </Modal>
  );
};
