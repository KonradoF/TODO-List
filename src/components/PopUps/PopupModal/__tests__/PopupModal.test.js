import React from 'react';
import {shallow} from 'enzyme';
import {PopupModal} from 'src/components/PopUps/PopupModal/PopupModal';

describe('PopupModal', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <PopupModal visible={true} onClose={jest.fn()} />,
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe('Check render props', () => {
    let props;
    let component;
    const mockFn = jest.fn();
    const visible = true;
    const difficulty = 2;
    const price = 1;
    const rating = '2.49';
    const time = '60';
    const animationType = 'fade';

    beforeEach(() => {
      component = shallow(
        <PopupModal
          onClose={mockFn}
          visible={visible}
          difficulty={difficulty}
          price={price}
          rating={rating}
          time={time}
        />,
      );
      props = component.props();
      jest.clearAllMocks();
    });

    it('should call onClose', () => {
      props.onRequestClose();
      expect(mockFn).toHaveBeenCalled();
    });
    it('should be visible', () => {
      expect(props.visible).toEqual(visible);
    });
    it('should match animationType', () => {
      expect(props.animationType).toEqual(animationType);
    });
  });
});
