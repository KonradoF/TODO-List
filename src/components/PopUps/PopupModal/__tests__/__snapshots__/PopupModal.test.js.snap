// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupModal Rendering should match to snapshot 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    onTouchStart={[MockFunction]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
      }
    }
  />
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "flex": 0.5,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "justifyContent": "space-between",
          "paddingBottom": 40,
          "paddingHorizontal": "5%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "width": "100%",
          }
        }
      />
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "backgroundColor": "#979797",
            "paddingBottom": 20,
            "paddingTop": 20,
          }
        }
        containerStyle={
          Object {
            "height": 120,
            "justifyContent": "center",
            "width": "100%",
          }
        }
        onPress={[MockFunction]}
        title="Close"
      />
    </View>
  </ForwardRef(SafeAreaView)>
</Component>
`;
